samples/node/routes/weather.js

```diff
@@ -9,19 +9,26 @@ const uuidV4 = require('uuid/v4');
 const sha1 = require('sha1');
 
 exports.requestCards = function(req, res) {
-    if (!req.body.tokens) {
-        res.status(400).send("Missing tokens field");
+    let tokenZip = (req.body.tokens && req.body.tokens.zip) ? req.body.tokens.zip : null;
+    let configZip = (req.body.config && req.body.config.defaultZip) ? [req.body.config.defaultZip] : null;
+
+    if (!tokenZip && !configZip) {
+        res.status(400).send("Missing zip.  Please provide one in tokens or config.");
         return;
     }
 
-    // Treat missing zips and empty zips the same way
-    const zips = req.body.tokens.zip || [];
+    console.log(`tokenZip: ${tokenZip}`);
+    console.log(`configZip: ${configZip}`);
+
+    const zips = tokenZip ? tokenZip : configZip;
```

Comment:
There's no reasy for a ternary here.  Just do this:

```
const zips = tokenZip || configZip;
```

Reply:
These are not booleans, they are the actual objects. This doesn't work for actual objects right?

Reply:
Javascript is a dynamically + weakly typed language.  It doesn't care if it's a boolean or not, only that it's truthy/falsy.

Also, you said above that it's the arrays (it is), but now you're saying objects.  Usually you don't call arrays objects in javascript (they are, however it's usually -- but not always -- important to distinguish the difference).